src/npctrade.cpp

```diff
@@ -67,244 +108,322 @@ std::vector<item_pricing> init_selling( npc &p )
+                auto color = it == &person.weapon ? c_yellow : c_light_gray;
+                const int &owner_sells = they ? ip.u_has : ip.npc_has;
+                const int &owner_sells_charge = they ? ip.u_charges : ip.npc_charges;
+                std::string itname = it->display_name();
+                if( ip.loc.where() != item_location::type::character ) {
+                    itname = itname + " " + ip.loc.describe( &g->u );
+                    color = c_light_blue;
+                }
+                if( ip.charges > 0 && owner_sells_charge > 0 ) {
+                    itname += string_format( _( ": trading %d" ), owner_sells_charge );
+                } else {
+                    if( ip.count > 1 ) {
+                        itname += string_format( _( " (%d)" ), ip.count );
+                    }
+                    if( owner_sells ) {
+                        itname += string_format( _( ": trading %d" ), owner_sells );
+                    }
                 }
-            }
 
-            for( auto &pricing : theirs ) {
-                if( pricing.selected ) {
-                    without.insert( pricing.loc.get_item() );
+                if( ip.selected ) {
+                    color = c_white;
                 }
-            }
 
-            temp = inventory_exchange( p.inv, without, added );
-
-            volume_left = p.volume_capacity() - p.volume_carried_with_tweaks( { temp } );
-            weight_left = p.weight_capacity() - p.weight_carried_with_tweaks( { temp } );
-            mvwprintz( w_head, 3, 2, ( volume_left < 0_ml || weight_left < 0_gram ) ? c_red : c_green,
-                       _( "Volume: %s %s, Weight: %.1f %s" ),
-                       format_volume( volume_left ), volume_units_abbr(),
-                       convert_weight( weight_left ), weight_units() );
-
-            std::string cost_string = ex ? _( "Exchange" ) : ( cash >= 0 ? _( "Profit %s" ) :
-                                      _( "Cost %s" ) );
-            mvwprintz( w_head, 3, TERMX / 2 + ( TERMX / 2 - cost_string.length() ) / 2,
-                       ( cash < 0 && static_cast<int>( g->u.cash ) >= cash * -1 ) || ( cash >= 0 &&
-                               static_cast<int>( p.cash )  >= cash ) ? c_green : c_red,
-                       cost_string.c_str(), format_money( std::abs( cash ) ) );
-
-            if( !deal.empty() ) {
-                mvwprintz( w_head, 3, ( TERMX - deal.length() ) / 2, cost < 0 ? c_light_red : c_light_green,
-                           deal.c_str() );
+                int keychar = i - offset + 'a';
+                if( keychar > 'z' ) {
+                    keychar = keychar - 'z' - 1 + 'A';
+                }
+                trim_and_print( w_whose, i - offset + 1, 1, win_w, color, "%c %c %s",
+                                static_cast<char>( keychar ), ip.selected ? '+' : '-', itname );
+#if defined(__ANDROID__)
+                ctxt.register_manual_key( keychar, itname );
```

Comment:
No input context were defined previously, so we should probably add before `for` cycle:

```
#if defined(__ANDROID__)
        input_context ctxt( "NPC_TRADE" );
#endif
```